shakenfist/daemons/queues.py

```diff
@@ -381,8 +382,7 @@ def instance_delete(inst):
 def snapshot(inst, disk, artifact_uuid, blob_uuid):
     blob.snapshot_disk(disk, blob_uuid)
     a = Artifact.from_db(artifact_uuid)
-    if a.state == dbo.STATE_INITIAL:
-        a.state = dbo.STATE_CREATED
+    a.state = dbo.STATE_CREATED
```

Comment:
Would it be better to reference the class of the object? ie. not `dbo` but `Artifact.STATE_CREATED`.

It seems nit picky but when I've been searching for state assignments the code currently has both usages. One should be chosen and I would argue for attribute of the actual class.

Reply:
I think most places use the base object version. They do that because it felt more natural to use the generic form of the generic states. Given these are actually just strings in the code I don't think there's a way to enforce which you use.

What's the advantage of using the derived version?

Reply:
The advantage is that assignments of state for Artefact can easily be found. They are can be visually verified as correct.

Enforcement would be difficult. The solutions I can think of would require extra code when converting to/from JSON. We have that already for other cases but not ideal for such a minor benefit.

(Should we swap to golang?)

Reply:
I've filed https://github.com/shakenfist/shakenfist/issues/974 to track this.